flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java

```diff
@@ -149,4 +149,10 @@ TypedResult<Integer> snapshotResult(String sessionId, String resultId, int pageS
     Optional<String> stopJob(
             String sessionId, String jobId, boolean isWithSavepoint, boolean isWithDrain)
             throws SqlExecutionException;
+
+    /** Get current catalog name. */
+    String getCurrentCatalogName(String sessionId);
+
+    /** Get current database name. */
+    String getCurrentDatabaseName(String sessionId);
```

Comment:
from one side yes, however `getCurrentCatalogName` and `getCurrentDatabaseName` are not a part of default value. It means by default it will not invoke those methods. These methods will be invoked only once it is required for the prompt